### PR TITLE
man-db: add support for zstd + fix undeclared runtime deps

### PR DESCRIPTION
**Pull request by DavHau**
###### Motivation for this change
man-db supports zstd since 2.9.2 (1 June 2020).
There might be man pages out there which are compressed with this format.
Also this opens up the possibility to compress with zstd instead of gzip in nixpkgs to benefit from faster builds and lower closure sizes.

###### Things done
- Add zstd to nativeBuildInputs
- patch configure.ac to properly reference runtime dependencies via absolute paths

The package previously had undeclared runtime deps like gnugrep or gzip.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/man-db/default.nix [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
+, libpipeline
+, libtool
+, makeWrapper
+, pkg-config
+, stdenv
+, zstd
+}:
 
 stdenv.mkDerivation rec {
   name = "man-db-2.9.4";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/man-db/default.nix:20]: Maybe we can also switch to pname + version.

File: pkgs/tools/misc/man-db/default.nix [hunk 2]
```diff
@@ -43,7 +65,15 @@ stdenv.mkDerivation rec {
     "ac_cv_func_mempcpy=no"
   ];
 
+
   preConfigure = ''
+    # need to recreate configure script due to substitutions in postPatch
+    ./bootstrap --gnulib-srcdir=${gnulib.src} --no-git
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/man-db/default.nix:71]: Should we use autoReconfHook instead?

File: pkgs/tools/misc/man-db/default.nix [hunk 3]
```diff
@@ -43,7 +65,15 @@ stdenv.mkDerivation rec {
...
+
+    # deal with autoconf 2.70 bug: https://lists.gnu.org/archive/html/bug-autoconf/2020-12/msg00036.html
+    # can be removed once autoconf 2.71 is merged
+    patch < ${./fix-configure.patch}
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/man-db/default.nix:75]: Could this be in patches?

Comment (DavHau) [reply pkgs/tools/misc/man-db/default.nix:72]: I guess not. This fixes the invalid configure script generated by autoconf which happens after the patch phase.

**Comment by DavHau** [comment]
Switched to pname + version and utilizing autoreconfHook now.

File: pkgs/tools/misc/man-db/default.nix [hunk 4]
```diff
@@ -1,17 +1,32 @@
...
-    url = "mirror://savannah/man-db/${name}.tar.xz";
+    url = "mirror://savannah/man-db/${pname}-${version}.tar.xz";
     sha256 = "sha256-tmyZ7frRatkoyIn4fPdjgCY8FgkyPCgLOp5pY/2xZ1Y=";
   };
 
   outputs = [ "out" "doc" ];
   outputMan = "out"; # users will want `man man` to work
 
-  nativeBuildInputs = [ pkg-config makeWrapper groff ];
+  nativeBuildInputs = [ autoreconfHook gettext groff libtool makeWrapper pkg-config zstd ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/man-db/default.nix:29]:
```suggestion
  nativeBuildInputs = [ autoreconfHook groff makeWrapper pkg-config zstd ];
```

they are already included in the hook.

**Comment by DavHau** [comment]
@SuperSandro2000 Anything left to do here?

**Comment by happysalada** [comment]
happy to help you get this merged if you can rebase this.

**Comment by happysalada** [comment]
alright just tested on darwin, and zstd does end up in the dependents.
looks good to me.
Thanks!

**Comment by Mindavi** [comment]
When cross-compiling this now shows this error, causing the build to fail:

`Found an erroneous dependency on groff ^^^`.

This is an error from the derivation, written in the postFIxup.

Possibly due to this?
`--replace AC_CHECK_PROGS AC_PATH_PROGS`

Haven't looked further into this other than seeing it fail on my hydra instance, but could use some investigation.

**Comment by DavHau** [comment]
The current PR removed `groff` from the nativeBuildInputs. Maybe it should be added again.

**Comment by Mindavi** [comment]
Hmm, I don't think that's what happened since I still see groff in the nariveBuildInputs in staging-next

**Comment by Artturin** [comment]
https://github.com/NixOS/nixpkgs/pull/169380

**Pull request merged**